### PR TITLE
Error handling in sagas, connected-react-router and tests

### PR DESCRIPTION
**Pull request by ecl1pseo**

### CODE REVIEW

File: src/containers/navbar-left/navbar-left.js [hunk 1]
```diff
@@ -30,11 +25,16 @@ const NavbarLeft = () => {
     }
   };
 
-  const categoriesItems = categories.map(({ _id, name }) => (
-    <Link key={_id} className={styles.link} to={`/${getCategoryURL(name)}`}>
-      {name[LANGUAGE].value}
-    </Link>
-  ));
+  const CategoriesView = () => {
```
Comment (Alan18081) [inline src/containers/navbar-left/navbar-left.js:28]: Please, don't have two react components at the same file, it is confusing. If it is really big you can create for it a new folder and with standard component folder structure. If it is small and used only inside this component then, just revert these changes and add assign to the `categoriesItems` `null` if no categories have been fetched from the backend

File: src/pages/home/categories/categories.js [hunk 2]
```diff
@@ -14,14 +14,19 @@ const Categories = () => {
...
 
-  const items = categories.map(({ _id, name, images, categoryCode }) => (
-    <CategoryItem
-      key={_id}
-      categoryCode={categoryCode}
-      categoryName={name[LANGUAGE].value}
-      categoryImage={images.large}
-    />
-  ));
+  const CategoriesView = () => {
```
Comment (Alan18081) [inline src/pages/home/categories/categories.js:17]: Please, remove this component, it is really very useless one

File: src/redux/categories/categories.sagas.js [hunk 3]
```diff
@@ -1,31 +1,41 @@
...
     const categories = yield call(getItems, query);
     yield put(setCategories(categories.data.getAllCategories));
+    yield put(setCategoriesLoading(false));
   } catch (e) {
-    console.log(e);
+    yield call(handleCategoriesError, e);
   }
 }
 
+function* handleCategoriesError(e) {
```
Comment (Alan18081) [inline src/redux/categories/categories.sagas.js:33]: Why have you created a separate function for error handling if it is used only once?

File: src/redux/categories/tests/categories.reducer.test.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import categoriesReducer from '../categories.reducer';
+import { SET_CATEGORIES } from '../categories.types';
+
+describe('categories.reducer.test', () => {
```
Comment (Alan18081) [inline src/redux/categories/tests/categories.reducer.test.js:4]: In names of test cases, please, use capitalized plain words without dots

File: src/redux/categories/tests/categories.reducer.test.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      loading: true
+    };
+  });
+
+  it('should return default state', () => {
+    expect(categoriesReducer(initialState, {})).toEqual(initialState);
+  });
+
+  it('should return state with new categories', () => {
+    const newCategories = {
```
Comment (Alan18081) [inline src/redux/categories/tests/categories.reducer.test.js:18]: Please, create actions via action creators because frankly speaking reducer waits for the object created by the action creator

File: src/redux/categories/tests/categories.sagas.test.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            {
+              value: 'Bags',
+              lang: 'en'
+            }
+          ]
+        }
+      }
+    };
+
+    return expectSaga(handleCategoriesLoad)
```
Comment (Alan18081) [inline src/redux/categories/tests/categories.sagas.test.js:24]: Looks cool)))

File: src/redux/news/news.sagas.js [hunk 7]
```diff
@@ -82,11 +84,17 @@ function* handleArticleLoad({ payload }) {
     );
     yield put(setArticle(article.data.getNewsById));
     yield put(setLoading(false));
-  } catch (error) {
-    console.log(error);
+  } catch (e) {
+    yield call(handleNewsError, e);
```
Comment (Alan18081) [inline src/redux/news/news.sagas.js:88]: Why have you created a separate function for error handling if it is used only once?

File: src/redux/error/error.reducer.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import { SET_ERROR } from './error.types';
+
+const initialState = {
+  error: null
```
Comment (Alan18081) [inline src/redux/error/error.reducer.js:4]: How does this error reducer influence `UI`?

File: src/redux/news/news.sagas.js (hunk 7 above)
Comment (ecl1pseo) [reply src/redux/news/news.sagas.js:88]: In news.sagas it's used two times - in getAllNews and getNewsById

File: src/containers/footer-lists/footer-lists.js [hunk 9]
```diff
@@ -35,15 +29,21 @@ const FooterLists = ({ language = LANGUAGE }) => {
...
-  const categoriesList = categories.map(({ _id, name }) => (
-    <div key={_id}>
-      <Typography variant='subtitle2'>
-        <Link className={styles.cardLink} to={`/${getCategoryURL(name)}`}>
-          {name[language].value}
-        </Link>
-      </Typography>
-    </div>
-  ));
+  const CategoriesView = () => {
```
[the remark this answers is not in the document]
Comment (ecl1pseo) [reply src/containers/footer-lists/footer-lists.js]: Fixed

File: src/containers/navbar-left/navbar-left.js (hunk 1 above)
Comment (ecl1pseo) [reply src/containers/navbar-left/navbar-left.js]: Fixed

File: src/pages/home/categories/categories.js (hunk 2 above)
Comment (ecl1pseo) [reply src/pages/home/categories/categories.js]: Fixed

File: src/redux/categories/categories.sagas.js (hunk 3 above)
Comment (ecl1pseo) [reply src/redux/categories/categories.sagas.js]: Removed

File: src/redux/categories/tests/categories.reducer.test.js (hunk 4 above)
Comment (ecl1pseo) [reply src/redux/categories/tests/categories.reducer.test.js]: Renamed

File: src/redux/categories/tests/categories.reducer.test.js (hunk 5 above)
Comment (ecl1pseo) [reply src/redux/categories/tests/categories.reducer.test.js:18]: Created with action creaters

File: src/redux/error/error.reducer.js (hunk 8 above)
Comment (ecl1pseo) [reply src/redux/error/error.reducer.js:4]: This error reducer set error object to the store and after that error-page can view to user type of error in different languages.

File: src/containers/footer-lists/footer-lists.js [hunk 10]
```diff
@@ -29,11 +29,9 @@ const FooterLists = ({ language = LANGUAGE }) => {
     }
   };
 
-  const CategoriesView = () => {
-    if (!categories) {
-      return null;
-    }
-    return categories.map(({ _id, name }) => (
+  const categoriesList = !categories
```
Comment (Alan18081) [inline src/containers/footer-lists/footer-lists.js:32]: Usually, it is better to have positive case first `categories ? categories.map(...) : null`

File: src/containers/navbar-left/navbar-left.js [hunk 11]
```diff
@@ -25,16 +25,13 @@ const NavbarLeft = () => {
     }
   };
 
-  const CategoriesView = () => {
-    if (!categories) {
-      return null;
-    }
-    return categories.map(({ _id, name }) => (
+  const categoriesList = !categories
```
Comment (Alan18081) [inline src/containers/navbar-left/navbar-left.js:28]: Usually, it is better to have positive case first `categories ? categories.map(...) : null`

File: src/pages/home/categories/categories.js [hunk 12]
```diff
@@ -14,19 +14,16 @@ const Categories = () => {
   }));
   const styles = useStyles();
 
-  const CategoriesView = () => {
-    if (!categories) {
-      return null;
-    }
-    return categories.map(({ _id, name, images, categoryCode }) => (
+  const categoriesList = !categories
+    ? null
```
Comment (Alan18081) [inline src/pages/home/categories/categories.js:18]: Usually, it is better to have positive case first `categories ? categories.map(...) : null`

File: src/containers/footer-lists/footer-lists.js (hunk 10 above)
Comment (ecl1pseo) [reply src/containers/footer-lists/footer-lists.js]: Changed

File: src/containers/navbar-left/navbar-left.js (hunk 11 above)
Comment (ecl1pseo) [reply src/containers/navbar-left/navbar-left.js]: Changed

File: src/pages/home/categories/categories.js (hunk 12 above)
Comment (ecl1pseo) [reply src/pages/home/categories/categories.js]: Changed

File: cypress/integration/navbar.spec.js [hunk 13]
```diff
@@ -0,0 +1,27 @@
+describe('Navbar test', () => {
+  it('items should be visible', () => {
+    cy.visit('/');
+    cy.viewport(1280, 720);
+    cy.get('.makeStyles-root-2').should('be.visible');
+    cy.get('.makeStyles-logo-5').should('be.visible');
+    cy.get('.MuiToolbar-root > [href="/backpacks"]').should('be.visible');
+    cy.get('.MuiToolbar-root > [href="/backpacks"]').should('be.visible');
```
Comment (bandvov) [inline cypress/integration/navbar.spec.js:8]: I gess here should be not three same rows but backpacks, bags,accessories

Comment (ecl1pseo) [reply cypress/integration/navbar.spec.js]: Fixed
